server/tweetSearch.js

```diff
@@ -321,6 +323,28 @@ module.exports = function(client, fs, speakerFile) {
         });
     }
 
+    function addSpeaker(name) {
+        speakers.push(name);
+        writeToFile();
+    }
+
+    function removeSpeaker(name) {
+        if (speakers.indexOf(name) > -1) {
+            speakers.splice(speakers.indexOf(name), 1);
+            writeToFile();
+        } else {
+            console.log("ERROR : Speaker not found in the speakers list");
+        }
+    }
+
+    function writeToFile() {
+        fs.writeFile(speakerFile, JSON.stringify({"speakers" : speakers}), function(err) {
```

Comment:
Apparently you can get some bad behaviour if `fs.writeFile` is called twice before the first call finishes; it's probably not gonna cause any errors worth worrying about for now though, so you can just slap in a Jira bug/TODO comment or whatever for now.